partner_phone_validation/models/res_partner.py

```diff
@@ -0,0 +1,54 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, models
+
+
+class ResPartner(models.Model):
+
+    _name = 'res.partner'
+    _inherit = ['res.partner', 'phone.validation.mixin']
+
+    @api.onchange('phone', 'country_id', 'company_id')
+    def _onchange_phone_validation(self):
+        if self.phone:
+            self.phone = self.phone_format(self.phone)
+
+    @api.onchange('mobile', 'country_id', 'company_id')
+    def _onchange_mobile_validation(self):
+        if self.mobile:
+            self.mobile = self.phone_format(self.mobile)
+
+    @api.model
+    def _get_phone_fields(self):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/app/numigi/odoo-partner-addons/pullRequest?prid=4335966)